### DISCUSSION TITLE
How can I suppress error logging in a test that is designed to fail?

### DISCUSSION THREAD

**Discussion by sokulski** [feathersjs/feathers#3530] [Q&A] [answered]
Hello,

I have the following test that is meant to fail, as it tests that an error is thrown when a request is made to delete a user.

```
  it("prevents deleting users", async () => {
    const user = await app.service("users").create({
      password: "passwordpassword",
      pin: 1234,
      nameFirst: "Walt",
      nameLast: "Disney",
    });

    const originalConsoleError = console.error;
    console.error = () => {};

    try {
      await assert.rejects(
        async () => {
          await app.service("users").remove(user.id);
        },
        {
          code: 405,
          className: "method-not-allowed",
        }
      );
    } finally {
      console.error = originalConsoleError;
    }
  });
```

A ``MethodNotAllowed`` error is being thrown when I run tests now. The test passes, but the error makes the mocha output harder to read.

I tried to reassign console.error in the call, but that didn't work.

Is there a way to do what I'm trying to do?

**Comment by daffl** [comment, marked as answer]
When using the [Winston logger](https://github.com/winstonjs/winston) you can configure [logger.ts](https://github.com/feathersjs/feathers-chat/blob/dove/feathers-chat-ts/src/logger.ts) to the log level you need according to the usage[ in tests. For example to remove all transports you'd do

```ts
import { logger } from '../logger'

logger.clear()

// Or to temporarily disable logging
logger.level = false
```
